common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -1154,9 +1154,33 @@ public static StringSearch getStringSearch(
    * Returns the collation ID for the given collation name.
    */
   public static int collationNameToId(String collationName) throws SparkException {
+    // If collation name is given as a fully qualified name, extract the actual collation name as
+    // the last part of the [catalog].[schema].[collation_name] name.
+    long numDots = collationName.chars().filter(ch -> ch == '.').count();
+    if (numDots > 0) {
+      String[] nameParts = collationName.split("\\.");
+      // Currently only predefined collations are supported.
+      if (numDots != 2 || !CollationFactory.CATALOG.equalsIgnoreCase(nameParts[0]) ||
+          !CollationFactory.SCHEMA.equalsIgnoreCase(nameParts[1])) {
+        throw CollationFactory.Collation.CollationSpec.collationInvalidNameException(collationName);
```

Comment:
afaik, `collationInvalidNameException` forms closest suggestions based on collation name- perhaps we should adopt one of the two approaches:

- not pass the fully qualified name to `collationInvalidNameException`
- make sure `collationInvalidNameException` handles it properly

Reply:
or we can add a new exception for this purpose, something like: `collationInvalidFullyQualifiedNameException`

Reply:
one other approach is to make sure that the suggestions algorithm returns fully qualified collation names at all times (although this might seem confusing to the user)

Reply:
I think that would be confusing. We don't want to encourage fully qualified names. We want to tolerate them.

Reply:
then pass only the last part to our suggestions algorithm